server/controllers/documentController.js

```diff
@@ -2,181 +2,196 @@ import db from '../models';
 
 
 const DocumentController = {
-  createNewDocument(req, res) {
-    db.documents
-      .create({
-        title: req.body.title,
-        content: req.body.content,
-        access: req.body.access || 'public',
-        userId: req.decodedToken.userId,
-        userRoleId: req.decodedToken.roleId
-      })
-      .then((document) => {
-        res.status(201).send({
-          message: 'Document created successfully',
-          data: document
-        });
-      })
-      .catch((err) => {
-        res.status(400).send({
-          message: 'There was an error while creating the document', err
-        });
+  createNewDocument(request, response) {
+    if (!request.body.title || !request.body.content) {
+      return response.status(400).send({
+        message:
+        'The title or the content of the document is empty'
+      });
+    }
+    db.documents.findOne({ where: { title: request.body.title } })
+      .then((documentExists) => {
+        if (documentExists) {
+          return response.status(400).send({
+            message: 'This document already exists'
+          });
+        }
+        db.documents
+          .create({
+            title: request.body.title,
+            content: request.body.content,
+            access: request.body.access || 'public',
+            userId: request.decodedToken.userId,
+            userRoleId: request.decodedToken.roleId
+          })
+          .then((document) => {
+            response.status(201).send({
+              message: 'Document created successfully',
+              data: document
+            });
+          })
+          .catch((err) => {
+            response.status(400).send({
+              message: 'There was an error while creating the document', err
```

Comment:
I think it will be better if you send the exact error to the user of your API.  Sequelize ships with a very powerful validation library. You can even write your own custom error messages and define your own validation rules with it. Lastly, its a convention to have the exact error that was thrown in error.message that could be a better option to send to the user. [Link to sequelize validation docs](http://docs.sequelizejs.com/en/latest/docs/models-definition/#validations)